bcmp/packet.c

```diff
@@ -416,14 +416,18 @@ BmErr packet_remove(BcmpMessageType type) {
  */
 BmErr process_received_message(void *payload, uint32_t size) {
   BmErr err = BmEINVAL;
-  BcmpProcessData data;
+  BcmpProcessData data = {0};
   BcmpSequencedRequestCb cb = NULL;
   BcmpRequestElement *request_message = NULL;
   BcmpPacketCfg *cfg = NULL;
   void *buf = NULL;
 
   if (payload && PACKET.initialized) {
     buf = PACKET.cb.data(payload);
+    if (buf == NULL) {
+      bm_debug("Recieved BCMP message with no contents!\n");
+      return err;
+    }
```

Comment:
Okay so there is no way to test if payload.data is NULL because if this buf is not NULL then the `void ptr + sizeof(BcmpHeader)` will always return something valid.

So the best we can do is make sure the buf is not null.

Reply:
LGTM